src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -62,38 +67,74 @@ const SearchPage = () => {
             },
           };
         });
-      setResults(finalResults);
+      setResults({ type: 'author', searchResults: finalResults });
     },
   });
 
   // Hook that will re-render the page only if the state of searchText changes
   useEffect(() => {
-    executeSearch();
-  }, [searchText]);
+    async function searchPosts() {
+      const res = await fetch(`${telescopeUrl}/query?search=${searchText}`);
+      const searchResults = await res.json();
+      const postIds = searchResults.values.map((result) => result.id);
+      let posts = await Promise.all(postIds.map((id) => fetch(`${telescopeUrl}/posts/${id}`)));
+      posts = await Promise.all(posts.map((post) => post.json()));
+      setResults({ type: 'post', searchResults: posts });
+    }
+
+    if (filter === 'author') executeSearch();
+    else if (filter === 'post') searchPosts();
+  }, [searchText, filter, telescopeUrl, executeSearch]);
 
   // Displays one of three options depending on whether there is a search string, results and no results
   const displayResults = () => {
-    if (searchText.length === 0 && results.length < 1) {
+    if (searchText.length > 0 && loading) {
+      return (
+        <h1 className={classes.searchReply}>
+          <Spinner />
+        </h1>
+      );
+    }
+
+    if (searchText.length === 0) {
       return <h1 className={classes.searchReply}>No search terms entered</h1>;
     }
-    //  The initial state of results is going to be 0 and we can't map 0,
-    //  this will check if there are any results and return AuthorResult component
-    //  for each with feed guid as key
-    if (results.length > 0)
-      return results.map((result) => (
-        <AuthorResult key={result.id} author={result.author} post={result.post} />
-      ));
-    return <h1 className={classes.searchReply}>No results found</h1>;
+
+    if (results) {
+      //  If result type is author return AuthorResult component
+      //  for each with feed guid as key
+      if (results.type === 'author' && results.searchResults.length > 0) {
+        return results.searchResults.map((result) => (
+          <AuthorResult key={result.id} author={result.author} post={result.post} />
```

Comment:
Personally, I don't see the value in the `AuthorResult`.  Why don't we just show you all the `<Post>` components for this user?  To me, that would be more logical.

Probably in follow-up we should do an auto-complete for Author, so you start typing and it shows you the people who have in our system and you choose one.  That would make this less necessary.